orc/src/java/org/apache/orc/impl/HadoopShimsCurrent.java

```diff
@@ -59,4 +63,30 @@ public DirectDecompressor getDirectDecompressor(
         return null;
     }
   }
+
+  @Override
+  public ZeroCopyReaderShim getZeroCopyReader(FSDataInputStream in,
+                                              ByteBufferPoolShim pool
+                                              ) throws IOException {
+    return ZeroCopyShims.getZeroCopyReader(in, pool);
+  }
+
+  private final class FastTextReaderShim implements TextReaderShim {
+    private final DataInputStream din;
+
+    public FastTextReaderShim(InputStream in) {
+      this.din = new DataInputStream(in);
```

Comment:
Ok, the file stream is the same one as the DefaultDataReader, but that one is also not closed. I'll make the zerocopyshim closeable.